### PR TITLE
refactor(ListWrapper): drop forEach and removeLast

### PR DESCRIPTION
**Pull request by vicb**

### CODE REVIEW

File: modules/angular2/src/core/facade/collection.dart [hunk 1]
```diff
@@ -160,7 +157,6 @@ class ListWrapper {
     }
   }
```
Comment (IgorMinar) [inline modules/angular2/src/core/facade/collection.dart]: is `removeAt` still needed?

Comment (vicb) [inline modules/angular2/src/core/facade/collection.dart]: yep because we can `removeAt` anywhere

**Comment by IgorMinar** [comment]
lgtm, if we can remove `removeAt` now then please do it, otherwise merge as is.

**Comment by pkozlowski-opensource** [comment]
Yay @vicb ! Seeing so much code removed makes me very happy :-)

**Comment by vicb** [comment]
I'll try to remove more as a background task (submitting ts2dart PRs when needed).

**Pull request closed without merging**